Sage-Bionetworks/BridgePF
# Integrate Surveys with Mutable Schemas; relax rules for Mutable Schemas

Last code change needed to finish off https://sagebionetworks.jira.com/browse/BRIDGE-1289

This change integrate surveys with mutable schemas. We also relax some of the rules for mutable schemas to make them easier to work with.

Testing done:
- unit tests updated
- integ tests (survey, upload schema, upload)
- manual tests

Manual test cases:
- create a new survey schema
- compatible update to survey schema
- incompatible update to survey schema
- explicitly ask for a new schema rev for a survey schema
- attempt to modify a schema field with an incompatible change
- successfully modify a schema field with a compatible change
- attempt to add a required field without minAppVersion to a schema
- successfully add a required field with minAppVersion to a schema

Note: https://github.com/Sage-Bionetworks/BridgeIntegrationTests/pull/76 needs to be pushed out before we can check this in.

TODO: Additional integ tests to test the updated functionality.
